src/BingoGame.Indexer.CA/Processors/PlayedProcessor.cs

```diff
@@ -0,0 +1,89 @@
+using AElfIndexer.Client;
+using AElfIndexer.Client.Handlers;
+using AElfIndexer.Grains.State.Client;
+using Google.Protobuf.WellKnownTypes;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Portkey.Contracts.BingoGameContract;
+using Portkey.Contracts.CA;
+using BingoGame.Indexer.CA.Entities;
+using BingoGame.Indexer.CA.GraphQL;
+using Volo.Abp.ObjectMapping;
+
+namespace BingoGame.Indexer.CA.Processors;
+
+public class PlayedProcessor : BingoGameProcessorBase<Played>
+{
+    private readonly IAElfIndexerClientEntityRepository<BingoGameIndex, TransactionInfo> _bingoIndexRepository;
+    private readonly IAElfIndexerClientEntityRepository<BingoGameStaticsIndex, TransactionInfo> _bingoStaticsIndexRepository;
+    private readonly IObjectMapper _objectMapper;
+    public PlayedProcessor(ILogger<PlayedProcessor> logger,
+        IAElfIndexerClientEntityRepository<BingoGameIndex, TransactionInfo> bingoIndexRepository,
+        IAElfIndexerClientEntityRepository<BingoGameStaticsIndex, TransactionInfo> bingoStaticsIndexRepository,
+        IOptionsSnapshot<ContractInfoOptions> contractInfoOptions,
+        IObjectMapper objectMapper) :
+        base(logger,objectMapper,contractInfoOptions)
+    {
+        _bingoIndexRepository = bingoIndexRepository;
+        _objectMapper = objectMapper;
+        _bingoStaticsIndexRepository = bingoStaticsIndexRepository;
+    }
+
+    public override string GetContractAddress(string chainId)
+    {
+        return ContractInfoOptions.ContractInfos.First(c=>c.ChainId == chainId).BingoGameContractAddress;
+    }
+
+    protected override async Task HandleEventAsync(Played eventValue, LogEventContext context)
+    {   
+        if (eventValue.PlayerAddress == null || eventValue.PlayerAddress.Value == null)
+        {
+            return;
+        }
+
+        var index = await _bingoIndexRepository.GetFromBlockStateSetAsync(eventValue.PlayId.ToHex(), context.ChainId);
+        if (index != null)
+        {
+            return;
+        }
```

Comment:
Same as above should we throw exception if the index doesn't exist?

Reply:
noted, added